api/v1/keplerdependency_types.go

```diff
@@ -23,13 +23,94 @@ import (
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
+//Enums
+
+// +kubebuilder:validation:Enum=FooOne;FooTwo
+type ModelPipeline string
+
+// +kubebuilder:validation:Enum=enable;disable
+type Cgroupv2 string
+
+const (
+	// random list of pipelines
+	FooOnePipeline ModelPipeline = "FooOne"
+	FooTwoPipeline ModelPipeline = "FooTwo"
+	// enum for cgroup options
+	EnableCgroupv2  Cgroupv2 = "enable"
+	DisableCgroupv2 Cgroupv2 = "disable"
+)
+
+// Collector CRD Requirements
+
+type RatioMetrics struct {
+	Global string `json:"global,omitempty"`
+	Core   string `json:"core,omitempty"`
+	Uncore string `json:"uncore,omitempty"`
+	Dram   string `json:"dram,omitempty"`
+}
+
+type Sources struct {
+	Cgroupv2 Cgroupv2 `json:"cgroupv2,omitempty"`
+	Bpf      string   `json:"bpf,omitempty"`
+	Counters string   `json:"counters,omitempty"`
+	Kubelet  string   `json:"kubelet,omitempty"`
+}
+
+type CollectorSpec struct {
+	Image        string       `json:"image,omitempty"`
+	Sources      Sources      `json:"sources,omitempty"`
+	RatioMetrics RatioMetrics `json:"ratioMetrics,omitempty"`
+}
+
+// Estimater CRD Requirements
+
+type NodeSelector struct {
+	//not sure if this is correct
```

Comment:
you can use core [NodeSelectorTerms](https://pkg.go.dev/k8s.io/api/core/v1#NodeSelectorTerm) API to support more affinity rules